src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/WidgetPageObject.java

```diff
@@ -0,0 +1,56 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.widget;
+
+import com.wikia.webdriver.common.contentpatterns.MercuryMessages;
+import com.wikia.webdriver.common.core.api.ArticleContent;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.mercury.BasePageObject;
+
+import org.openqa.selenium.WebDriver;
+
+/**
+ * @ownership: Content X-Wing
+ */
+public abstract class WidgetPageObject extends BasePageObject {
+
+  protected String tagName;
+
+  protected WidgetPageObject(WebDriver driver) {
+    super(driver);
+  }
+
+  protected abstract boolean isTagLoadedOnMercury();
+
+  protected abstract boolean isTagLoadedOnOasis();
+
+  protected String getTagName() {
+    return tagName;
+  }
+
+  private String getArticleName() {
+    return tagName.substring(0, 1).toUpperCase() + tagName.substring(1).toLowerCase() + "Widget";
```

Comment:
article name and tag (<tag>) could be defined as fields in TwitterWidget class so it's more verbose